docs/guides/postinstall.md

```diff
@@ -142,6 +142,186 @@ blacklist cdc_mbim" >> /etc/modprobe.d/blacklist.conf'
+            ;;
+        "dnf")
+            echo "$1 install"
+            ;;
+        esac
+    }
+    
+    pm_uninstall() {
+        case "$1" in
+        "pacman")
+            echo "$1 -Rnsu"
+            ;;
+        "dnf")
+            echo "$1 remove"
+            ;;
+        esac
+    }
+    
+    echo "Checking package manager..."
+    PM=$(get_package_manager)
+    if [[ "$PM" == "NONE" ]]; then
+        echo "No valid package manager has been found..."
+        echo "Exiting!"
+        exit 1
+    else
+        echo "Found package maanger: $PM"
+    fi
+    
+    PM_INSTALL=$(pm_install "$PM")
+    PM_UNINSTALL=$(pm_uninstall "$PM")
+    
+    # Checking dependencies
+    echo "Cheking dependencies..."
+    
+    if ! command -v git &>/dev/null; then # GIT
+        echo "git could not be found! Installing..."
+        $PM_INSTALL git
+    else
+        echo "git is installed..."
+    fi
+    
+    if ! command -v dkms &>/dev/null; then # DKMS
+        echo "dkms could not be found! Installing..."
+        $PM_INSTALL dkms
+    else
+        echo "dkms is installed..."
+    fi
+    
+    if [[ "$1" == "--old" ]]; then
+    
+        # Switching to old driver
+        echo "Switching to old driver..."
+    
+        # Checking if tiny-dfr is uninstalled
+        echo "Uninstalling tiny-dfr..."
+        if command -v tiny-dfr &>/dev/null; then
+            echo "tiny-dfr has been detected... Uninstalling!"
+            $PM_UNINSTALL tiny-dfr
+        else
+            echo "tiny-dfr is not installed..."
+        fi
+    
+        # Blacklisting new driver
+        echo "Blacklisting new touchbar driver..."
+    
+        sh -c 'echo "# Disable new Apple Touchbar driver
+    blacklist hid_appletb_kbd
+    blacklist hid_appletb_bl" > /etc/modprobe.d/touchbar.conf'
+    
+        if [ ! -d "/usr/src/apple-touchbar-0.1" ]; then
+    
+            # Cloning repo
+            echo "Cloning driver repo..."
+            git clone https://github.com/AdityaGarg8/apple-touchbar-drv /usr/src/apple-touchbar-0.1
+    
+        fi
+    
+        if [[ ! "$(dkms status)" == *"apple-touchbar"* ]]; then
+            # Installing driver
+            echo "Installing driver"
+            dkms install -m apple-touchbar -v 0.1
+        fi
+    
+    elif [[ "$1" == "--new" ]]; then
+    
+        # Switching to new driver
+        echo "Switching to new driver..."
+    
+        # Checking if tiny-dfr is uninstalled
+        echo "Installing tiny-dfr..."
+        if ! command -v tiny-dfr &>/dev/null; then
+            echo "tiny-dfr has not been detected... Installing!"
+            $PM_INSTALL tiny-dfr
+        else
+            echo "tiny-dfr is installed..."
+        fi
+    
+        # Blacklisting new driver
+        echo "Blacklisting old touchbar driver..."
+    
+        sh -c 'echo "# Disable new Apple Touchbar driver
+    blacklist apple-touchbar" > /etc/modprobe.d/touchbar.conf'
+    
+        # Cloning repo
+        echo "Cloning driver repo..."
```

Comment:
Why are you installing the driver if we want the new one?

Reply:
Fixed